### PR TITLE
[processing] Monkey patch stable external Processing API into qgis.processing

### PR DESCRIPTION
**Pull request by nyalldawson**
Instead of encouraging use of the internal Processing modules
(e.g. from processing.tools.general import run , import processing, ...)
instead expose all Python specific STABLE processing additions
to the qgis.processing module.

Instead, scripts and plugins should now use

from qgis.processing import run, algorithmHelp,...

This makes a clear distinction between internal Processing python
modules (i.e., everything else!) and the parts of Processing
which are stable and designed to be used by plugins and scripts.

TODO: QGIS 4.0 -- move the internal Processing plugin modules
to __processing, to clearer indicate that this is all internal
stuff.

### CODE REVIEW

File: python/plugins/processing/__init__.py [hunk 1]
```diff
@@ -31,6 +31,17 @@
...
+# monkey patch Python specific Processing API into stable qgis.processing module
+import qgis.processing
+qgis.processing.algorithmHelp = algorithmHelp
+qgis.processing.run = run
+qgis.processing.runAndLoadResults = runAndLoadResults
+qgis.processing.createAlgorithmDialog = createAlgorithmDialog
+qgis.processing.execAlgorithmDialog = execAlgorithmDialog
+qgis.processing.createContext = createContext
+qgis.processing.isWindows = isWindows
+qgis.processing.isMac = isMac
```
Comment (m-kuhn) [inline python/plugins/processing/__init__.py]: Is this processing specific?

**Comment by m-kuhn** [comment]
Good move!!!

**Comment by m-kuhn** [comment]
Backport to 3.4 too, right?

**Comment by nyalldawson** [comment]
I don't think qgis.processing is backported to 3.4 yet, but I will when that is!

**Comment by Gustry** [comment]
What about updating the script template `processing.run` ? https://github.com/qgis/QGIS/blob/master/python/plugins/processing/script/ScriptTemplate.py#L21

File: python/plugins/processing/__init__.py (hunk 1 above)
Comment (nyalldawson) [reply python/plugins/processing/__init__.py]: Nope, dropping

**Comment by nyalldawson** [comment]
@Gustry done!

**Pull request merged**